copy/single.go

```diff
@@ -593,6 +622,20 @@ type diffIDResult struct {
 	err    error
 }
 
+func (ic *imageCopier) getCompressionFromBlob(srcInfo types.BlobInfo) *compressiontypes.Algorithm {
+
+	if srcInfo.CompressionAlgorithm == nil {
```

Comment:
If we have that data, we should return it, not ignore it. Or clearly document that the caller needs to handle that case, and actually do that in the new caller.

Reply:
Looking at more details, only manifest’s `.LayerInfos()` never sets this field; only `pkg/blobcache` sets it in `LayerInfosForCopy`.

So the new caller in `compareImageDestinationManifestEqual` can be build assuming `==nil` above; and in that case this function can a simpler `compressionAlgorithmFromMIMEType`, leaving the `if` only in the `copyLayer` caller.